src/app/features/tech-record/components/tech-record-summary-changes/tech-record-summary-changes.component.spec.ts

```diff
@@ -0,0 +1,220 @@
+    });
+  });
+
+  describe('submit', () => {
+    it('should call the store 3 times', () => {
+      jest.spyOn(store, 'dispatch');
+      component.submit();
+      // eslint-disable-next-line @typescript-eslint/unbound-method
+      expect(store.dispatch).toHaveBeenCalledTimes(3);
+      // eslint-disable-next-line @typescript-eslint/unbound-method
+      expect(store.dispatch).toHaveBeenCalledWith({
+        systemNumber: '123456',
+        createdTimestamp: '123123123',
+        type: '[Technical Record Service] updateTechRecords',
+      });
+      // eslint-disable-next-line @typescript-eslint/unbound-method
+      expect(store.dispatch).toHaveBeenCalledWith({
+        type: '[Technical Record Service] clearAllSectionState',
+      });
+      // eslint-disable-next-line @typescript-eslint/unbound-method
+      expect(store.dispatch).toHaveBeenCalledWith({
+        type: '[Technical Record Service] clearScrollPosition',
+      });
+    });
+  });
+
+  describe('cancel', () => {
+    it('should call globalErrorService.clearErrors and then navigate', () => {
+      jest.spyOn(component.globalErrorService, 'clearErrors');
+      jest.spyOn(component.router, 'navigate');
+      component.cancel();
+      // eslint-disable-next-line @typescript-eslint/unbound-method
+      expect(component.globalErrorService.clearErrors).toHaveBeenCalled();
+      // eslint-disable-next-line @typescript-eslint/unbound-method
+      expect(component.router.navigate).toHaveBeenCalled();
+    });
+  });
+
+  describe('getTechRecordChangesKeys', () => {
+    it('should return a list of all of the populated keys', () => {
+      component.techRecordChanges = { techRecord_grossEecWeight: 1, techRecord_grossDesignWeight: 1 };
+      const keys = component.getTechRecordChangesKeys();
+      expect(keys).toEqual(['techRecord_grossEecWeight', 'techRecord_grossDesignWeight']);
+    });
+  });
+
+  describe('getSectionsWhitelist', () => {
+    it('should return an empty array if vehicleType is null', () => {
+      component.techRecordEdited = undefined;
+      const value = component.getSectionsWhitelist();
+      expect(value).toEqual([]);
+    });
+    it('should return an empty array if techRecordChanges is null', () => {
+      component.techRecord = undefined;
+      const value = component.getSectionsWhitelist();
+      expect(value).toEqual([]);
+    });
+    it('should call haveAxlesChanged if vehicleType and techRecordChanges are defined', () => {
+      const localTechRecordEdited = getEmptyTechRecord();
+      component.techRecordEdited = localTechRecordEdited as TechRecordType<'put'>;
+      component.techRecordChanges = { techRecord_grossEecWeight: 1, techRecord_grossDesignWeight: 1 };
+      const value = component.getSectionsWhitelist();
+      expect(value).toEqual(['weightsSection']);
+    });
+  });
+
+  describe('isNotEmpty', () => {
+    it('should return true if an object is populated', () => {
```

Comment:
This test feels like it dosnt belong here, is it the responsibility of this component to check this ?

feels like a utility method.